jarviscli/plugins/reminder.py

```diff
@@ -380,13 +560,87 @@ class Todo_Progress(TodoBase):
 
     def __call__(self, jarvis, s):
         entry = self.select_one_remind(jarvis)
-        entry['progress'] = jarvis.input("Progress: ")
+        inp = jarvis.input("Set a progress level between 0 and 100, as a percentage: ").replace("%", "")
+        if inp.isnumeric() and 0 <= int(inp) <= 100:
+            entry['progress'] = int(inp)
+        else:
+            jarvis.say("The progress level must be an integer between 0 and 100.", color=Fore.MAGENTA)
+    
         self.modify(jarvis, entry)
 
 
+@plugin('todo tag')
+class Todo_Tag(TodoBase):
+    """
+    Add a tag to an item which lets you filter later.
+    """
+    def __call__(self, jarvis, s):
+        entry = self.select_one_remind(jarvis)
+        tags = TagBase()
+        selected_tag = tags.select_one_tag(jarvis)
+        try:
+            entry["tags"].append(selected_tag["id"])
+        except KeyError:
+            entry["tags"] = [selected_tag["id"]]
+
+        self.modify(jarvis, entry)
+
+
+@plugin('todo del tag')
+class Todo_Del_Tag(TodoBase):
+    """
+    Remove a tag from a todo item.
+    """
+    def __call__(self, jarvis, s):
+        self.remove_tag(jarvis, s)
+
+
+@plugin('todo filter')
+class Todo_Filter(TodoBase):
+    """
+    Filter items on your todo list by tag.
+    """
+    def __call__(self, jarvis, s):
+        tags = TagBase()
+        selected_tag = tags.select_one_tag(jarvis)
+        self.do_print(jarvis, should_print=lambda entry: selected_tag["id"] in entry["tags"])
+
+
+@plugin('tags')
+class Tags(TagBase):
+    """
+    List currently created tags.
+    """
+    def __call__(self, jarvis, s):
+        self.do_print(jarvis)
+
+@alias('tags create', 'tags add')
+@plugin('tags new')
+class Tags_New(TagBase):
+    """
+    Create a new tag.
+    """
+    def __call__(self, jarvis, s):
+        self.add_tag(jarvis, s)
```

Comment:
Maybe check s != '' - because empty string tags (created when just calling tags create) might be confusing ;).